pdl_syntax.md

```diff
@@ -0,0 +1,819 @@
+---
+layout: guide
+title: PDL Syntax
+permalink: /pdl_syntax
+excerpt: Rest.li PDL Syntax.
+---
+
+# PDL Syntax
+
+## Contents
+-   [PDL Schema Definition](#pdl__schema_definition)
+-   [Creating a Schema](#creating_a_schema)
+-   [Record Type](#record_type)
+-   [Primitive Types](#primitive_type)
+-   [Array Type](#array_type)
+-   [Map Type](#map_type)
+-   [Union Type](#union_type)
+-   [Enum Type](#enum_type)
+-   [Fixed Type](#fixed_type)
+-   [Typerefs](#typerefs)
+-   [Namespace](#namespace\)
+-   [Import](#import)
+-   [Deprecation](#deprecation)
+-   [Package](#package)
+
+## PDL Schema Definition
+
+PDL is a Pegasus schema definition language, which is developer friendly and less verbose than the JSON based schema format.
+
+## Creating a Schema
+
+Each schema should be stored in its own file with a `.pdl` extension.
+The Pegasus code generator implements a resolver that is similar to Java
+class loaders. If there is a reference to a named schema, the code
+generator will try to look for a file in the code generator’s resolver
+path. The resolver path is similar to a Java classpath. The fully
+qualified name of the named schema will be translated to a relative file
+name. The relative file name is computed by replacing dots (“.”) in the
+fully qualified name by the directory path separator (typically “/”) and
+appending a `.pdl` extension. This relative file name is appended to
+each path in the resolver path. The resolver opens each of these files
+until it finds a file that contains the named schema.
+
+Pegasus supports different types of schemas: [Records](#record_type), [Primitive types](#record_type), [Enums](#enum_type), [Arrays](#array_type), [Maps](#map_type), [Unions](#union_type), [Fixed](#fixed_type) and [Typerefs](#typerefs). 
+Please check the following documentations for details.
+
+
+## Record Type
+Records contain any number of fields, which can be primitive types, enums, unions, maps and arrays.
+
+A basic record type can contain a few fields.
+
+For example:
+```
+import org.example.time.DateTime
+
+record Example {
+  field1: string
+  field2: optional int
+  field3: DateTime
+}
+```
+Record fields can be optional.
+
+For example:
+```
+namespace com.linkedin.pegasus.generator.examples
+
+/**
+ * A foo record
+ */
+record Foo {
+   field1: string
+   field2: optional string
```

Comment:
Add a doc to explain this field is optional.

Reply:
fixed